src/fromager/resolver.py

```diff
@@ -179,10 +179,10 @@ def get_preference(
     def is_satisfied_by(self, requirement: Requirement, candidate: Candidate) -> bool:
         if canonicalize_name(requirement.name) != candidate.name:
             return False
-        return (
-            candidate.version in requirement.specifier
-            and self.constraints.is_satisfied_by(requirement.name, candidate.version)
-        )
+        allow_prerelease = self.constraints.allow_prerelease(requirement.name)
```

Comment:
So this means we would normally not allow pre-release versions, unless a constraint is in place that mentions a pre-release version?

Reply:
yep